### PR TITLE
Mypy on blockchain, encoding, storage, network

### PR DESCRIPTION
**Pull request by pirapira**
Four new directories are covered by mypy checks.

The most significant change is the introduction of a new type
``UDPMessageID`` for udp_utils.py.  This was necessary because
``MessagerID`` was used for very different two types (one was ``int``
but the other was a tuple).

Besides, some type annotations were adjusted as usual.

### CODE REVIEW

File: raiden/raiden_service.py [hunk 1]
```diff
@@ -687,7 +687,7 @@ def install_all_blockchain_filters(
             )
 
             for token_network in token_networks:
-                token_network_proxy = self.chain.token_network(token_network)
+                token_network_proxy = self.chain.token_network(typing.Address(token_network))
```
Comment (pirapira) [inline raiden/raiden_service.py]: I don't know what's the best way to proceed here.  The callee looks like:

```
class TokenNetwork:
    def __init__(
            self,
            jsonrpc_client,
            manager_address, # <---- So it's just an Address?
            contract_manager: ContractManager,
    ):
```

File: raiden/utils/typing.py [hunk 2]
```diff
@@ -57,6 +57,8 @@
 T_MessageID = int
 MessageID = NewType('MessageID', T_MessageID)
 
+UDPMessageID = Tuple[str, int, Address]
```
Comment (LefterisJP) [inline raiden/utils/typing.py]: note that there is going to be a part in the code where type would be a union between `MessageID` and `UDPMessageID` but we can't do anything to avoid it.

File: raiden/raiden_service.py (hunk 1 above)
Comment (LefterisJP) [reply raiden/raiden_service.py]: Ah yes. This is a legacy name that got left over. `manager_address` refers to the old `ChannelManager` contract we had in the past. In this case you can go ahead and rename it to `token_network_address` as this is what this is. You can verify what I am saying by looking at what the constructor of `TokenNetwork` does with this address.

TL;DR: Change type to `TokenNetworkAddress`

File: raiden/utils/typing.py (hunk 2 above)
Comment (pirapira) [reply raiden/utils/typing.py]: I'm curious.

File: raiden/raiden_service.py (hunk 1 above)
Comment (pirapira) [reply raiden/raiden_service.py]: Done.  This PR is now 80 lines bigger.  That's a good thing.

File: raiden/utils/typing.py (hunk 2 above)
Comment (pirapira) [reply raiden/utils/typing.py]: I will see when it happens.

**Comment by pirapira** [comment]
@LefterisJP merge while the test is good?
